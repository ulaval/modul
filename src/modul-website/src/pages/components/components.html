<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
