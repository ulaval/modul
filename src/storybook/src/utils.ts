export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
