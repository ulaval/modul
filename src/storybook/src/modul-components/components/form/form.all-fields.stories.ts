import { FORM_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { FormControl } from '@ulaval/modul-components/dist/utils/form/form-control';
import { FormGroup } from '@ulaval/modul-components/dist/utils/form/form-group';
import { MaxLengthValidator } from '@ulaval/modul-components/dist/utils/form/validators/max-length/max-length';
import { RequiredValidator } from '@ulaval/modul-components/dist/utils/form/validators/required/required';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { FormBuilder } from './helpers/form-builder';

const COUPE_NAMES: string[] = ['régulière', 'julienne', 'ondulé'];
const TYPE_NAMES: string[] = ['douce', 'blanche', 'sec'];

export default {
    title: `${modulComponentsHierarchyRootSeparator}${FORM_NAME}/all fields`,
    parameters: { fileName: __filename }
};

export const textfield = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'name' }),
                        MaxLengthValidator(20, { controlLabel: 'name' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-textfield v-m-control="control"
                         v-model="control.value"
                         label="text field"
                         :error="control.hasError()"
                         :error-message="control.errorMessage"">
            </m-textfield>
        </template>
    </form-builder>`
});



export const textarea = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'description': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'description' }),
                        MaxLengthValidator(255, { controlLabel: 'description' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
        <m-textarea v-m-control="control"
                v-model="control.value"
                label="Description Field"
                :error="control.hasError()"
                :error-message="control.errorMessage""></m-textarea>
        </template>
    </form-builder>`
});

export const datepicker = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'birthDate': new FormControl<string>(
                    [
                        RequiredValidator()
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-datepicker v-m-control="control"
                          v-model="control.value"
                          label="Birthdate"
                          min="1900-01-01"
                          max="2020-01-01"
                          :required-marker="true"
                          :error="control.hasError()"
                          :error-message="control.errorMessage"></m-datepicker>
        </template>
    </form-builder>`
});

export const dropdown = () => ({
    components: { FormBuilder },
    data: () => ({
        types: [...TYPE_NAMES],
        formGroup: new FormGroup(
            {
                'type': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'type' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-dropdown v-m-control="control"
                        v-model="control.value"
                        label="Type"
                        :required-marker="true"
                        :error="control.hasError()"
                        :error-message="control.errorMessage">
                <m-dropdown-item v-for="type of types"
                                :value="type"
                                :label="type"></m-dropdown-item>
            </m-dropdown>
        </template>
    </form-builder>`
});

export const checkbox = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'active': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'active' })
                    ]
                )
            }
        )
    }),
    template: `
        <form-builder :form-group="formGroup">
            <template v-slot="{ control }">
                <m-checkbox v-m-control="control"
                            v-model="control.value"
                            :error="control.hasError()"
                            :error-message="control.errorMessage">Is active ?</m-checkbox>
            </template>
        </form-builder>`

});

export const radio = () => ({
    components: { FormBuilder },
    data: () => ({
        coupes: [...COUPE_NAMES],
        formGroup: new FormGroup(
            {
                'coupe': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'Style de coupe' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-radio-group v-m-control="control"
                        v-model="control.value"
                        label="Style de coupe"
                        :error="control.hasError()"
                        :error-message="control.errorMessage">
                <m-radio v-for="coupe of coupes"
                    :value="coupe">{{coupe}}</m-radio>
                </m-radio-group>
        </template>
    </form-builder>`
});

export const timepicker = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'time': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'Time' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
        <m-timepicker v-m-control="control"
                      v-model="control.value"
                      label="Time"
                      :required-marker="true"
                      :error="control.hasError()"
                      :error-message="control.errorMessage"></m-timepicker>
        </template>
    </form-builder>`
});

export const decimalfield = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'decimal': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'decimal' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-decimalfield v-m-control="control"
                            v-model="control.value"
                            label="Decimal"
                            :required-marker="true"
                            :error="control.hasError()"
                            :error-message="control.errorMessage">

            </m-decimalfield>
        </template>
    </form-builder>`
});

export const moneyfield = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'money': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'Price' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-moneyfield v-m-control="control"
                        v-model="control.value"
                        label="Price"
                        :required-marker="true"
                        :error="control.hasError()"
                        :error-message="control.errorMessage"></m-moneyfield>
        </template>
    </form-builder>`

});

export const integerfield = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'integer': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'Integer' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-integerfield v-m-control="control"
                            v-model="control.value"
                            label="Integer"
                            :required-marker="true"
                            :error="control.hasError()"
                            :error-message="control.errorMessage">

            </m-integerfield>
        </template>
    </form-builder>`
});

export const switchs = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'switch': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'switch' })
                    ]
                )
            }
        )
    }),
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-switch v-m-control="control"
                    v-model="control.value"
                    :error="control.hasError()"
                    :error-message="control.errorMessage">État</m-switch>
        </template>
    </form-builder>`
});

export const autocomplete = () => ({
    components: { FormBuilder },
    data: () => ({
        autocompleteResults: [{ label: 'RandomDog', value: 'RandomDog' }, {
            label: 'RandomDog2',
            value: 'RandomDog2'
        }],
        formGroup: new FormGroup(
            {
                'autocomplete': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'autocomplete' })
                    ]
                )
            }
        )
    }),
    methods: {
        onComplete(): void {
            this.$data.autocompleteResults = [...this.$data.autocompleteResults];
        }
    },
    computed: {
        autocompleteField(): void {
            return (this).$data.formGroup.getControl('autocomplete');
        }
    },
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-autocomplete v-m-control="control"
                            v-model="control.value"
                            label="Autocomplete"
                            :results="autocompleteResults"
                            :required-marker="true"
                            :error="control.hasError()"
                            :error-message="control.errorMessage"
                            @complete="onComplete"></m-autocomplete>
        </template>
    </form-builder>`
});

export const daterangepicker = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'daterange': new FormControl<string>(
                    [
                        RequiredValidator({ controlLabel: 'daterange' })
                    ]
                )
            }
        )
    }),
    computed: {
        daterangeField(): void {
            return (this).$data.formGroup.getControl('daterange');
        }
    },
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-daterangepicker v-m-control="control"
                            v-model="control.value"
                            :error="control.hasError()"
                            :error-message="control.errorMessage">
            </m-daterangepicker>
        </template>
    </form-builder>`
});

export const multiSelect = () => ({
    components: { FormBuilder },
    data: () => ({
        formGroup: new FormGroup(
            {
                'multi-select': new FormControl<string[]>(
                    [
                        RequiredValidator({ controlLabel: 'multi-select' })
                    ], { initialValue: [] }
                )
            }
        ),
        options: ['Maria Rainer', 'Georg von Trapp', 'Liesl von Trapp', 'Friedrich von Trapp', 'Louisa von Trapp', 'Kurt von Trapp', 'Brigitta von Trapp', 'Marta von Trapp', 'Gretl von Trapp']
    }),
    computed: {
        multiSelectField(): void {
            return (this).$data.formGroup.getControl('multi-select');
        }
    },
    template: `
    <form-builder :form-group="formGroup">
        <template v-slot="{ control }">
            <m-multi-select v-model="control.value"
                            v-m-control="control"
                            :error="control.hasError()"
                            :error-message="control.errorMessage"
                            label="La famille Von Trapp"
                            :link-select-all="true"
                            :options="options"
                            :required-marker="true">
            </m-multi-select>
        </template>
    </form-builder>`
});
