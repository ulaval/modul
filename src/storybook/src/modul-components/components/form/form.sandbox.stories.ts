import { FORM_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { sandboxDecorator } from '../../../sandbox-decorator';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${FORM_NAME}/sandboxes`,
    decorators: [sandboxDecorator],
    parameters: { fileName: __filename, options: { showPanel: false, isToolshown: true } }
};

export const complex: any = () => `<${FORM_NAME}-complex-sandbox></${FORM_NAME}-complex-sandbox>`;

export const externalValidation: any = () => `<${FORM_NAME}-external-validation-sandbox></${FORM_NAME}-external-validation-sandbox>`;

export const reactivity: any = () => `<${FORM_NAME}-reactivity-sandbox></${FORM_NAME}-reactivity-sandbox>`;
