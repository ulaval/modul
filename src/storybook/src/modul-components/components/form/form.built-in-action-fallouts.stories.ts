import { FORM_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { ClearErrorToast, ClearSummaryMessage, ErrorToast, FocusOnFirstError, SummaryMessage } from '@ulaval/modul-components/dist/components/form/fallouts/built-in-form-action-fallouts';
import { FormControl } from '@ulaval/modul-components/dist/utils/form/form-control';
import { FormGroup } from '@ulaval/modul-components/dist/utils/form/form-group';
import { EmailValidator } from '@ulaval/modul-components/dist/utils/form/validators/email/email';
import { MaxLengthValidator } from '@ulaval/modul-components/dist/utils/form/validators/max-length/max-length';
import { RequiredValidator } from '@ulaval/modul-components/dist/utils/form/validators/required/required';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${FORM_NAME}/built-in action-fallouts`,
    parameters: { fileName: __filename }
};


export const defaultFallout = () => ({

    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>(
                    [RequiredValidator({ controlLabel: 'name' })]
                )
            }
        )
    }),
    computed: {
        nameField(): void {
            return this.$data.formGroup.getControl('name');
        }
    },
    template: `
        <m-form class="m-u--margin-top"
                :form-group="formGroup">
            <m-textfield v-model.trim="nameField.value"
                        :error-message="nameField.errors.length > 0 ? nameField.errors[0].message : null"
                        :label="nameField.name"
                        v-m-control="nameField">
            </m-textfield>
            <p class="m-u--margin-bottom--l">
                <m-button type="submit">Submit</m-button>
                <m-button type="reset"
                            skin="secondary">Reset</m-button>
            </p>
        </m-form>
        `
});

export const toastAndClear = () => ({

    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>(
                    [
                        RequiredValidator()
                    ]
                ),
                'email': new FormControl<string>(
                    [
                        EmailValidator(),
                        MaxLengthValidator(10)
                    ]
                )
            }
        ),
        actionFallouts: [
            ErrorToast,
            ClearErrorToast
        ]
    }),
    computed: {
        nameControl(): FormControl<string> {
            return this.$data.formGroup.getControl('name') as FormControl<string>;
        },
        emailControl(): FormControl<string> {
            return (this).$data.formGroup.getControl('email') as FormControl<string>;
        }
    },
    template: `
        <m-form class="m-u--margin-top"
                :form-group="formGroup"
                :action-fallouts="actionFallouts">
            <m-textfield v-model.trim="nameControl.value"
                        :error="nameControl.hasError()"
                        :error-message="nameControl.errorMessage"
                        :label="nameControl.name"
                        v-m-control="nameControl">
            </m-textfield>
            <m-textfield v-model.trim="emailControl.value"
                        :error="emailControl.hasError()"
                        :error-message="emailControl.errorMessage"
                        :label="emailControl.name"
                        v-m-control="emailControl">
            </m-textfield>
            <p class="m-u--margin-bottom--l">
                <m-button type="submit">Submit</m-button>
                <m-button type="reset"
                            skin="secondary">Reset</m-button>
            </p>
        </m-form>
        `
});

export const focusFirstError = () => ({

    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>(
                    [RequiredValidator({ controlLabel: 'name' })]
                )
            }
        ),
        actionFallouts: [
            FocusOnFirstError
        ]
    }),
    computed: {
        nameField(): void {
            return (this).$data.formGroup.getControl('name');
        }
    },
    template:
        ` <m-form class="m-u--margin-top"
                :form-group="formGroup"
                :action-fallouts="actionFallouts">
            <m-textfield v-model.trim="nameField.value"
                        :error-message="nameField.errors.length > 0 ? nameField.errors[0].message : null"
                        :label="nameField.name"
                        v-m-control="nameField">
            </m-textfield>
            <p class="m-u--margin-bottom--l">
                <m-button type="submit">Submit</m-button>
                <m-button type="reset"
                            skin="secondary">Reset</m-button>
            </p>
        </m-form>`

});

export const messageSummaryAndClear = () => ({

    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>(
                    [RequiredValidator({ controlLabel: 'name' })]
                )
            }
        ),
        actionFallouts: [
            SummaryMessage,
            ClearSummaryMessage
        ]
    }),
    computed: {
        nameField(): void {
            return (this).$data.formGroup.getControl('name');
        }
    },
    template: `
        <m-form class="m-u--margin-top"
                :form-group="formGroup"
                :action-fallouts="actionFallouts">
            <m-textfield v-model.trim="nameField.value"
                        :error-message="nameField.errors.length > 0 ? nameField.errors[0].message : null"
                        :label="nameField.name"
                        v-m-control="nameField">
            </m-textfield>
            <p class="m-u--margin-bottom--l">
                <m-button type="submit">Submit</m-button>
                <m-button type="reset"
                            skin="secondary">Reset</m-button>
            </p>
        </m-form>
        `
});
