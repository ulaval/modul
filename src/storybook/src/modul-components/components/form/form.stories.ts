import { FORM_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { ClearErrorToast, ErrorToast } from '@ulaval/modul-components/dist/components/form/fallouts/built-in-form-action-fallouts';
import { FormControl } from '@ulaval/modul-components/dist/utils/form/form-control';
import { FormGroup } from '@ulaval/modul-components/dist/utils/form/form-group';
import { CompareValidator } from '@ulaval/modul-components/dist/utils/form/validators/compare/compare';
import { MinLengthValidator } from '@ulaval/modul-components/dist/utils/form/validators/min-length/min-length';
import { RequiredValidator } from '@ulaval/modul-components/dist/utils/form/validators/required/required';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${FORM_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    methods: {
        submit: () => {
            Vue.prototype.$log.log('submited');
        },
        reset: () => {
            Vue.prototype.$log.log('reseted');
        }
    },
    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>()
            }
        )
    }),
    computed: {
        nameField(): void {
            return this.$data.formGroup.getControl('name');
        }
    },
    template: `
        <m-form class="m-u--margin-top"
                :form-group="formGroup"
                @reset="reset()"
                @submit="submit()">
            <h2>Default with no validations</h2>
            <m-textfield v-model.trim="nameField.value"
                        :error-message="nameField.errors.length > 0 ? nameField.errors[0].message : null"
                        :label="nameField.name"
                        v-m-control="nameField">
            </m-textfield>
            <p class="m-u--margin-bottom--l">
                <m-button type="submit">Submit</m-button>
                <m-button type="reset"
                            skin="secondary">Reset</m-button>
            </p>
        </m-form>
        `
});

defaultStory.story = {
    name: 'default'
};

export const submitOutsideForm = () => ({
    methods: {
        submit: () => {
            Vue.prototype.$log.log('submited');
        },
        reset: () => {
            Vue.prototype.$log.log('reseted');
        }
    },
    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>()
            }
        )
    }),
    template: `
        <div>
            <m-form class="m-u--margin-top"
                id="123-456"
                :form-group="formGroup"
                @reset="reset()"
                @submit="submit()">
            <m-textfield v-model.trim="formGroup.getControl('name').value"
                        :error-message="formGroup.getControl('name').errors.length > 0 ? formGroup.getControl('name').errors[0].message : null"
                        :label="formGroup.getControl('name').name"
                        v-m-control="formGroup.getControl('name')">
            </m-textfield>
            </m-form>
            <p class="m-u--margin-bottom--l">
                <m-button type="submit"
                        form="123-456">Submit</m-button>
                <m-button type="reset"
                        skin="secondary"
                        form="123-456">Reset</m-button>
            </p>
        </div>

        `
});

export const reactiveInitialValue = () => ({
    data: () => ({
        formGroup: new FormGroup(
            {
                'name': new FormControl<string>([RequiredValidator({ controlLabel: 'name' })], { initialValue: 'blabla' })
            }

        )
    }),
    methods: {
        submit(value: string): void {
            Vue.prototype.$log.log('submited');
            this.$data.formGroup.getControl('name').initalValue = value;
        },
        reset(): void {
            Vue.prototype.$log.log('reseted');
        },
        resetToNewValue(): void {
            this.$data.formGroup.getControl('name').reset('newValue');
            Vue.prototype.$log.log('resetToNewValue');
        }

    },
    template: `
            <div>
                <m-form class="m-u--margin-top"
                    :form-group="formGroup"
                    @reset="reset()"
                    @submit="submit(formGroup.getControl('name').value)">
                <m-textfield v-model.trim="formGroup.getControl('name').value"
                            :error-message="formGroup.getControl('name').errors.length > 0 ? formGroup.getControl('name').errors[0].message : null"
                            :label="formGroup.getControl('name').name"
                            v-m-control="formGroup.getControl('name')">
                </m-textfield>
                <p class="m-u--margin-bottom--l">
                    <m-button type="submit">Submit</m-button>
                    <m-button type="reset"
                            >Reset</m-button>
                    <m-button skin="secondary"
                            @click="resetToNewValue">Reset to new value</m-button>
                </p>
                </m-form>
            </div>
            `
});

export const toastMessageCountForFormGroup = () => ({
    data: () => ({
        formGroup: new FormGroup({
            'control1': new FormControl<string>([RequiredValidator()]),
            'control2': new FormControl<string>([RequiredValidator()]),
            'control3': new FormControl<string>([RequiredValidator()]),
            'formGroup1': new FormGroup(
                {
                    'nestedControl1': new FormControl<string>([MinLengthValidator(5)]),
                    'nestedControl2': new FormControl<string>([MinLengthValidator(5)])
                },
                [
                    CompareValidator([
                        'nestedControl1', 'nestedControl2'
                    ])
                ]
            ),
            'formGroup2': new FormGroup(
                {
                    'nestedControl1': new FormControl<string>([MinLengthValidator(5)]),
                    'nestedControl2': new FormControl<string>([MinLengthValidator(5)])
                },
                [
                    CompareValidator([
                        'nestedControl1', 'nestedControl2'
                    ])
                ]
            )
        }),
        actionFallouts: [
            ErrorToast,
            ClearErrorToast
        ]
    }),
    computed: {
        nameControl(): FormControl<string> {
            return this.$data.formGroup.getControl('name') as FormControl<string>;
        },
        emailControl(): FormControl<string> {
            return (this).$data.formGroup.getControl('email') as FormControl<string>;
        }
    },
    template: `
        <m-form :form-group="formGroup"
            @reset="reset()"
            @submit="submit()">
        <h4 class="m-u--h6">General</h4>
        <m-textfield v-for="(control, index) in formGroup.controls.slice(0, 3)"
                    :key="index + 'A'"
                    v-m-control="control"
                    v-model="control.value"
                    label="test"
                    :required-marker="true"
                    :error="control.hasError()"
                    :error-message="control.errorMessage">
        </m-textfield> <br />
        <hr />
        <m-input-group :error="formGroup.getControl('formGroup1').hasError()"
                    :error-message="formGroup.getControl('formGroup1').errorMessage">
            <m-textfield v-for="(control, index) in formGroup.getControl('formGroup1').controls"
                        :key="index  + 'B'"
                        v-m-control="control"
                        v-model="control.value"
                        label="test"
                        helper-message="5 characters min"
                        :error="control.hasError()"
                        :error-message="control.errorMessage">
            </m-textfield> <br />
        </m-input-group>
        <m-input-group :error="formGroup.getControl('formGroup2').hasError()"
                    :error-message="formGroup.getControl('formGroup2').errorMessage">
            <m-textfield v-for="(control, index) in formGroup.getControl('formGroup2').controls"
                        :key="index  + 'C'"
                        v-m-control="control"
                        v-model="control.value"
                        label="test"
                        helper-message="5 characters min"
                        :error="control.hasError()"
                        :error-message="control.errorMessage">
            </m-textfield> <br />
        </m-input-group>
        <p>
            <m-button type="submit">submit</m-button>
            <m-button type="reset">reset</m-button>
        </p>
    </m-form>
    `
});
