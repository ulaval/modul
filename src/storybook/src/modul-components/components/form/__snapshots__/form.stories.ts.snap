// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-form Default Story 1`] = `
<form
  class="m-form m-u--margin-top"
  novalidate="novalidate"
>
  <!---->
   
  <!---->
   
  <h2>
    Default with no validations
  </h2>
   
  <div
    class="m-textfield m--is-type-text"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <span
            class="m-input-style__label"
            style="z-index: -1;"
          >
            <span
              class="m-input-style__text"
            />
          </span>
           
          <div
            class="m-input-style__input"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__content"
            >
              <input
                class="m-textfield__input"
                id="mTextfield-fakeId"
                maxlength="Infinity"
                type="text"
              />
                 
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
              <!---->
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textfield__validation"
    >
      <div
        class="m-validation-message m-textfield__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
   
  <p
    class="m-u--margin-bottom--l"
  >
    <button
      class="m-button m--is-skin-primary"
      type="submit"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Submit 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <button
      class="m-button m--is-skin-secondary"
      type="reset"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Reset 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
  </p>
</form>
`;

exports[`Storyshots modul-components|m-form Reactive Initial Value 1`] = `
<div>
  <form
    class="m-form m-u--margin-top"
    novalidate="novalidate"
  >
    <!---->
     
    <!---->
     
    <div
      class="m-textfield m--is-type-text"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--has-value m--is-tag-default"
        style="margin-top: 10px;"
      >
        <div
          class="m-input-style__main"
        >
          <div
            class="m-input-style__body"
          >
            <span
              class="m-input-style__label"
              style="z-index: -1;"
            >
              <span
                class="m-input-style__text"
              />
            </span>
             
            <div
              class="m-input-style__input"
            >
              <div
                class="m-input-style__prefix"
              />
               
              <div
                class="m-input-style__content"
              >
                <input
                  class="m-textfield__input"
                  id="mTextfield-fakeId"
                  maxlength="Infinity"
                  type="text"
                />
                   
                <!---->
                 
                <!---->
                 
                <!---->
              </div>
               
              <div
                class="m-input-style__suffix"
              >
                <!---->
                 
                <!---->
              </div>
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <div
        class="m-textfield__validation"
      >
        <div
          class="m-validation-message m-textfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
    </div>
     
    <p
      class="m-u--margin-bottom--l"
    >
      <button
        class="m-button m--is-skin-primary"
        type="submit"
      >
        <!---->
         
        <!---->
         
        <span
          class="m-button__text"
        >
          Submit 
          <!---->
        </span>
         
        <!---->
         
        <!---->
      </button>
       
      <button
        class="m-button m--is-skin-primary"
        type="reset"
      >
        <!---->
         
        <!---->
         
        <span
          class="m-button__text"
        >
          Reset 
          <!---->
        </span>
         
        <!---->
         
        <!---->
      </button>
       
      <button
        class="m-button m--is-skin-secondary"
        type="button"
      >
        <!---->
         
        <!---->
         
        <span
          class="m-button__text"
        >
          Reset to new value 
          <!---->
        </span>
         
        <!---->
         
        <!---->
      </button>
    </p>
  </form>
</div>
`;

exports[`Storyshots modul-components|m-form Submit Outside Form 1`] = `
<div>
  <form
    class="m-form m-u--margin-top"
    id="123-456"
    novalidate="novalidate"
  >
    <!---->
     
    <!---->
     
    <div
      class="m-textfield m--is-type-text"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-tag-default"
        style="margin-top: 10px;"
      >
        <div
          class="m-input-style__main"
        >
          <div
            class="m-input-style__body"
          >
            <span
              class="m-input-style__label"
              style="z-index: -1;"
            >
              <span
                class="m-input-style__text"
              />
            </span>
             
            <div
              class="m-input-style__input"
            >
              <div
                class="m-input-style__prefix"
              />
               
              <div
                class="m-input-style__content"
              >
                <input
                  class="m-textfield__input"
                  id="mTextfield-fakeId"
                  maxlength="Infinity"
                  type="text"
                />
                   
                <!---->
                 
                <!---->
                 
                <!---->
              </div>
               
              <div
                class="m-input-style__suffix"
              >
                <!---->
                 
                <!---->
              </div>
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <div
        class="m-textfield__validation"
      >
        <div
          class="m-validation-message m-textfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
    </div>
  </form>
   
  <p
    class="m-u--margin-bottom--l"
  >
    <button
      class="m-button m--is-skin-primary"
      form="123-456"
      type="submit"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Submit 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <button
      class="m-button m--is-skin-secondary"
      form="123-456"
      type="reset"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Reset 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
  </p>
</div>
`;
